activerecord/lib/active_record/railtie.rb

```diff
@@ -26,6 +26,9 @@ class Railtie < Rails::Railtie # :nodoc:
     config.active_record.use_schema_cache_dump = true
     config.active_record.maintain_test_schema = true
 
+    config.active_record_sqlite3adapter = ActiveSupport::OrderedOptions.new
+    config.active_record_sqlite3adapter.represent_boolean_as_integer = nil
```

Comment:
Yes, other `set_configs` do not use `respond_to`.  
It is exceptionally used by the Action Controller because it has a config set to `ActionController::Base`.

Action Controller: https://github.com/rails/rails/blob/master/actionpack/lib/action_controller/railtie.rb#L58..L62 
Action Mailer: https://github.com/rails/rails/blob/master/actionmailer/lib/action_mailer/railtie.rb#L47 
Action Cable: https://github.com/rails/rails/blob/master/actioncable/lib/action_cable/engine.rb#L37 
Active Job: https://github.com/rails/rails/blob/master/activejob/lib/active_job/railtie.rb#L20 
Action View: https://github.com/rails/rails/blob/master/actionview/lib/action_view/railtie.rb#L36